pkg/gatewayserver/gatewayserver.go

```diff
@@ -950,3 +949,11 @@ func (gs *GatewayServer) UpdateConnectionStats(conn *io.Connection) error {
 func (gs *GatewayServer) ClearConnectionStats(conn *io.Connection) error {
 	return gs.statsRegistry.Set(conn.Context(), conn.Gateway().GatewayIdentifiers, nil)
 }
+
+func cloneGatewayUplinkMessage(msg *ttnpb.GatewayUplinkMessage) *ttnpb.GatewayUplinkMessage {
```

Comment:
And not in `pkg/gatewayserver` because future devs might assume that it is cloning the argument while it is not